Lib/idlelib/autocomplete.py

```diff
@@ -137,18 +313,35 @@ def open_completions(self, args):
                     return None
             else:
                 comp_what = ""
+        elif mode == DICTKEYS:
+            pass
         else:
             return None
 
         if complete and not comp_what and not comp_start:
             return None
         comp_lists = self.fetch_completions(comp_what, mode)
+        if mode == DICTKEYS:
+            assert comp_lists[0] == comp_lists[1]
+            reprs = self._dict_key_reprs(comp_start, comp_lists[0])
+
+            if hp.is_in_code():
+                # Add variables in the global namespace to the comp lists.
+                comp_lists = (reprs, list(reprs))
+                extra_comp_lists = self.fetch_completions("", ATTRS)
```

Comment:
This should be deleted.

Reply:
Please see my comment above re. intentionally including global variables in the lists.

Reply:
I will try to explain better in a post to the issue why I am currently opposed to this.